types/jest/index.d.ts

```diff
@@ -604,8 +609,10 @@ declare namespace jest {
         /**
          * Matches any array made up entirely of elements in the provided array.
          * You can use it inside `toEqual` or `toBeCalledWith` instead of a literal value.
+         *
+         * Optionally, you can provide a type for the elements via a generic.
          */
-        arrayContaining(arr: any[]): any;
+        arrayContaining<E>(arr: E[]): any;
```

Comment:
```suggestion
        arrayContaining<E = any>(arr: E[]): any;
```

Explicit fallback rather than implicit